tendrl/gluster_integration/objects/volume/atoms/check_volume_available/__init__.py

```diff
@@ -0,0 +1,52 @@
+import etcd
+import gevent
+import subprocess
+
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons import objects
+from tendrl.commons.objects import AtomExecutionFailedError
+from tendrl.gluster_integration.objects.volume import Volume
+
+
+class CheckVolumeAvailable(objects.BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(CheckVolumeAvailable, self).__init__(*args, **kwargs)
+
+    def run(self):
+        retry_count = 0
+        while True:
+            try:
+                volumes = NS._int.client.read(
+                    "clusters/%s/Volumes" % NS.tendrl_context.integration_id
+                )
+            except etcd.EtcdKeyNotFound:
+                # ignore as no volumes available till now
+                pass
+
+            if volumes:
+                for entry in volumes.leaves:
+                    volume = Volume(vol_id=entry.key.split("Volumes/")[-1]).load()
+                    if volume.name == self.parameters['Volume.volname']:
+                        return True
+
+            retry_count += 1
+            gevent.sleep(1)
```

Comment:
Increase the timeout to 10 mins, we are going to allow configurable sds/node sync intervals in near future